vulnfeeds/cves/versions.go

```diff
@@ -71,30 +141,55 @@ func ExtractVersionInfo(cve CVEItem, validVersions []string) VersionInfo {
 		}
 	}
 
+	gotVersions := false
 	for _, node := range cve.Configurations.Nodes {
 		if node.Operator != "OR" {
 			continue
 		}
 
-		// TODO: Also try to parse description as these are not always reliably set.
 		for _, match := range node.CPEMatch {
 			if !match.Vulnerable {
 				continue
 			}
 
-			if match.VersionStartIncluding != "" || match.VersionEndExcluding != "" {
-				if match.VersionStartExcluding != "" || match.VersionEndIncluding != "" {
-					// TODO: handle these by using validVersions.
-					continue
-				}
-
-				v.AffectedVersions = append(v.AffectedVersions, AffectedVersion{
-					// TODO: make sure these match validVersions.
-					Introduced: match.VersionStartIncluding,
-					Fixed:      match.VersionEndExcluding,
-				})
+			introduced := ""
+			fixed := ""
+			if match.VersionStartIncluding != "" {
+				introduced = match.VersionStartIncluding
+			} else if match.VersionStartExcluding != "" {
+				introduced = nextVersion(validVersions, match.VersionStartExcluding)
```

Comment:
if the cpe match is not a valid version, then this returns the first version valid version. Want to confirm this is by design.

for e.g. the vuln started way before and now there are only new valid versions -> all are affected.

Reply:
As above this is wrong and fixed.